etcdserver/etcdhttp/peer.go

```diff
@@ -19,15 +19,21 @@ import (
 	"net/http"
 
 	"github.com/coreos/etcd/etcdserver"
+	"github.com/coreos/etcd/lease"
 	"github.com/coreos/etcd/rafthttp"
 )
 
 const (
 	peerMembersPrefix = "/members"
+	leasesPrefix      = "/leases"
 )
 
-// NewPeerHandler generates an http.Handler to handle etcd peer (raft) requests.
-func NewPeerHandler(cluster etcdserver.Cluster, raftHandler http.Handler) http.Handler {
+// NewPeerHandler generates an http.Handler to handle etcd peer requests.
+func NewPeerHandler(s *etcdserver.EtcdServer) http.Handler {
+	return newPeerHandler(s.Cluster(), s.RaftHandler(), s.Lessor())
+}
+
+func newPeerHandler(cluster etcdserver.Cluster, raftHandler http.Handler, lessor lease.Lessor) http.Handler {
```

Comment:
can we change lessor type arg to lessorHandler for this handler func? And then create the lessor Handler in `NewPeerHandler` func at line 33?